geo_tag/geo_tag.py

```diff
@@ -0,0 +1,200 @@
+import json
+import os
+import pickle
+
+from geopy.geocoders import Nominatim
+
+class TwitterJSONTagger():
+    def __init__(self):
+        # data is the tweet json which we add geo_tag.
+        self.data = dict()
+        # cache is the hashmap { (city, state), other geo info}
+        self.cache = dict()
+        # coord is the (longitude, latitude)
+        self.coord = tuple()
+        # self.abbrev_us_state is the relation between the state's full name and abbrev.
+        with open('us_state_abbrev.json', 'r') as inf:
+            self.abbrev_us_state = eval(inf.read())
+
+    def init_geo_cache(self, city_json):
+        # check pickle module
+        filename = "cache_geo_hashmap"
+        if os.path.exists(filename):
+            cache_file = open(filename, 'rb')
+            self.cache = pickle.load(cache_file)
+            return
+        else:
+            feature = city_json.get('features')
+            for f in feature:
+                item = f.get('properties')
+                # store the city, state as key; (cityID, countyID, stateID, countyName) as value.
+                if item["name"] is not None and item["stateName"] is not None:
+                    _city_state = item["name"] + ", " + item["stateName"]
+                    # key, value
+                    self.cache[_city_state] = (item["cityID"], item["countyID"], item["stateID"], item["countyName"])
+
+                else:
+                    print("[ERROR] city.json data.")
+                    return
+
+            # store it using pickle
+            cache_file = open(filename, 'wb')
+            pickle.dump(self.cache, cache_file)
+
+    def get_coordinate(self):
```

Comment:
The API of this method is not clear. `get_coordinate` infers this method will have coord as a return value. but instead it returns None and assigns the coord to self. Please make it return coord instead and assign the value in outside scope.
i.e.,
```
self.coord = self.get_coordinate()
```